tests/test_xopen.py

```diff
@@ -721,3 +721,20 @@ def test_valid_compression_levels(writer, level, tmpdir):
     with writer(test_file, "wb", level) as handle:
         handle.write(b"test")
     assert gzip.decompress(Path(test_file).read_bytes()) == b"test"
+
+
+def test_format_override(tmp_path):
```

Comment:
Hm, I think also a test is needed that forces an incorrect input format, which then should raise an error (forcing gzip but reading uncompressed, for example).

Reply:
I agree. Added it.